### PR TITLE
don't allow setting cvar on module that's frozen

### PR DESCRIPTION
**Pull request by luke-gru**

### CODE REVIEW

File: test/ruby/test_variable.rb [hunk 1]
```diff
@@ -155,6 +155,20 @@ def test_include_refined_module_class_variable
...
+    end
+  end
+
+  def test_set_cvar_on_frozen_object
+    SetCvar.set(1) # fill write cache
+    SetCvar.freeze
+    assert_raise(FrozenError) do
+      SetCvar.set(2) # hit write cache, but should check frozen status
+    end
+  end
```
Comment (nobu) [inline test/ruby/test_variable.rb:170]:
```suggestion
  def test_set_cvar_on_frozen_object
    set_cvar = EnvUtil.labeled_class("SetCVar") do
      def self.set(val)
        @@a = val # inline cache
      end
    end

    set_cvar.set(1) # fill write cache
    set_cvar.freeze
    assert_raise(FrozenError) do
      set_cvar.set(2) # hit write cache, but should check frozen status
    end
  end
```

Comment (luke-gru) [reply test/ruby/test_variable.rb:170]: Oh yeah that's cool, and it'll be GC'd like any old object. Thanks!

**Comment by luke-gru** [comment]
Ran into issue using EnvUtil.labeled class with class variables: https://bugs.ruby-lang.org/issues/19345

File: test/ruby/test_variable.rb [hunk 2]
```diff
@@ -155,6 +155,19 @@ def test_include_refined_module_class_variable
     end
   end
 
+  def test_set_class_variable_on_frozen_object
+    set_cvar = EnvUtil.labeled_class("SetCVar") do
+      def self.set(val)
+        @@a = val # inline cache
+      end
+    end
```
Comment (nobu) [inline test/ruby/test_variable.rb:163]: `class_eval` is needed.
```suggestion
    set_cvar = EnvUtil.labeled_class("SetCVar")
    set_cvar.class_eval "#{<<~"begin;"}\n#{<<~'end;'}"
    begin;
      def self.set(val)
        @@a = val # inline cache
      end
    end;
```

Comment (luke-gru) [reply test/ruby/test_variable.rb:163]: So this pushes cref? Thanks!

**Review by peterzhu2118 — COMMENTED** [review]
Can you tag your commit with a `[Bug #19341]` so that the ticket will automatically be linked to the commit when merged?

**Comment by luke-gru** [comment]
Done :)

**Pull request merged**